src/Config/Loader/XliffFileLoader.php

```diff
@@ -76,11 +76,40 @@ private function convertXlfToPhp($name, $language)
         $xml = $this->getDomDocumentFromFile($name);
 
         $return = "\n// ".str_replace(strtr(dirname($this->rootDir), '\\', '/').'/', '', strtr($name, '\\', '/'))."\n";
-        $units = $xml->getElementsByTagName('trans-unit');
+        $fileNodes = $xml->getElementsByTagName('file');
+        $language = strtolower($language);
+
+        /** @var \DOMElement[] $fileNodes */
+        foreach ($fileNodes as $fileNode) {
+            $tagName = 'target';
+
+            // Use the source tag if the source language matches
+            if (strtolower($fileNode->getAttribute('source-language')) === $language) {
+                $tagName = 'source';
+            }
```

Comment:
IMO we should add a check here if either the `target-language` or the `source-language` match and if neither matches, return empty.
It is entirely possible to have something with source "de" and target "fr" and now the content from "fr" will be used on the "en" page which IMO should be prevented.
The above cited use case could happen when multiple entries have been combined into one big XLIFF file where each originates from a different translator.